src/steps/EnterAmount/index.tsx

```diff
@@ -103,25 +103,28 @@ const EnterAmount: React.FC<Props> = () => {
   ]);
 
   useEffect(() => {
-    const base = parseFloat(formValue.base || '0');
```

Comment:
Consider the following values
`chainBalance.sub(fee)`: 73336768223643846 (wei)
`maxLimitFormatted`: 0.073336768223643846 (ether)

In JS the max number of decimals is 17, and parseFloat will round up. In that case:
`base`: 0.07333676822364385

`0.07333676822364385` will get parsed to ether which results in a wei value greater than your max spendable.

`parseFloat` is not safe and should not be used. @plondon look into changing https://github.com/map3xyz/payments-sdk/pull/85/files#diff-2afda4e5b582778b536f39923cc84b0ab2dfec5f3e73fb00c1fdc749b53f9180R74